tooling/image-updater/update.sh

```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+set -euo pipefail
+
+CREATE_PR=true
+COMPONENTS=""
+EXCLUDE_COMPONENTS=""
+
+if [ -n "${GITHUB_SERVER_URL:-}" ] && [ -n "${GITHUB_REPOSITORY:-}" ] && [ -n "${GITHUB_RUN_ID:-}" ]; then
+    WORKFLOW_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}"
+    AUTOMATION_CREDIT="PR created automatically with [Image Digest Updater](${WORKFLOW_URL})"
+else
+    AUTOMATION_CREDIT="Automatically updated with Image Digest Updater"
+fi
+
+log() {
+    echo "[$(date +'%H:%M:%S')] $1" >&2
+}
+
+usage() {
+    cat << EOF
+Usage: $0 [OPTIONS]
+
+Bulk update all component image digests and create a single PR.
+
+OPTIONS:
+    --components NAMES          Update only specified components (comma-separated)
```

Comment:
I don't mean to be snarky, but by the time you are developing usage patterns, flag parsing, and a logging utility it should be exceedingly obvious that this is not a good use of Shell. Please check in a nice CLI that automates this.